Review DomInstances bulk error handling and update examples [ID 41546] (#4014)
SkylineCommunications/dataminer-docs: user-guide/Advanced_Modules/DOM/DOM_examples/DOM_BulkProcessing_Example.md

Author: On line 7 *DataMiner 10.4.2/10.5.0 [RN37891]* is mentioned. But these changes apply to *10.5.2/10.5.0 onwards [RN41546]*.
It will become too complicated to explain the differences here, so I would suggest mentioning the new version only since the main release is expected to be the same.

From DataMiner 10.4.2/10.5.0 onwards<!-- RN 37891 -->, the `DomInstance` CRUD helper component supports processing multiple `DomInstances` in one call.

In DataMiner 10.5.0/10.5.2<!-- RN 41546 -->, the error handling has been reviewed. This page contains examples on how you can use these calls, provided by the [DomHelper](xref:DomHelper_class#multiple-instances).

> [!IMPORTANT]
> In DataMiner versions prior to DataMiner Feature Release 10.5.0/10.5.2<!-- RN 37891 -->, when any validation issue occurs, no exception is thrown when calling the `CreateOrUpdate` or `Delete` methods. Instead, the result of the call should be used to check for which `DomInstances` the call succeeded or failed.

> > The `TryCreateOrUpdate` or `TryDelete` methods are not available in those versions.

Reviewer: Does "those versions" refer to DataMiner versions prior to 10.5.0/10.5.2 or to something else?

Author: Yes. They're getting added in RN41546

### Checking issues

> For some of the `DomInstances`, the creation or updating might not succeed. When using `CreateOrUpdate` or `Delete`, an exception will be thrown. That exception will include how many items succeeded and how many items failed for that bulk CRUD operation. Next to that a limited list of object IDs that succeeded is added. Also a list of object IDs that failed, each followed by the related TraceData. After that the default exception info gets added, including the stacktrace.

Author: Added a list after the example.

### Checking issues

For some of the `DomInstances`, the creation or update might not succeed. When using `CreateOrUpdate` or `Delete`, an exception will be thrown. Like in the previous example, some `DomInstances` need to be updated:

```csharp
try
{

>   // Update the DomInstances, similarly as in the above example.

Reviewer 2: ```suggestion
  // Update the DomInstances, similar to the above example.
```

}

// Log what items were successfully removed.

> Log($"Could perform the update successfully for {updateResult?.SuccessfulItems.Count ?? 0} items");

Reviewer 3: Feels a bit weird to need a null check on the updateResult here 🫤 Maybe we should add an if & check the tracedata on the CrudHelperComponent?

Author: That's in the next example that belongs to *Unexpected issue* (line 112). Just wanted this example to be correct. We could drop the other example and explain it here.

Author: Will combine the two examples.
